File: suites/02_numerical_computation/comparisons/fft.go
package main
import "fmt"
func butterflyCost(size int64) int64 {
	var total int64
	stage := int64(1)
	for stage < size {
		for i := int64(0); i < size; i += stage * 2 {
			left := ((i * 19) + stage) % 65521
			right := ((i * 23) + (stage * 3)) % 65521
			total += (left + right) - (left - right)
		}
		stage *= 2
	}
	return total
}
func main() { fmt.Println(butterflyCost(16384)) }

File: suites/02_numerical_computation/comparisons/fft.rs
fn butterfly_cost(size: i64) -> i64 {
    let mut total: i64 = 0;
    let mut stage: i64 = 1;
    while stage < size {
        let mut i: i64 = 0;
        while i < size {
            let left = ((i * 19) + stage) % 65521;
            let right = ((i * 23) + (stage * 3)) % 65521;
            total += (left + right) - (left - right);
            i += stage * 2;
        }
        stage *= 2;
    }
    total
}
fn main() { println!("{}", butterfly_cost(16384)); }
